### ISSUE TITLE
can not run tox in windows env

### ISSUE THREAD

**Issue by ws2003gq** [pypa/sampleproject#136]
the error below,how can i fix it ?

C:\python_test\sampleproject>python3 -m tox
GLOB sdist-make: C:\python_test\sampleproject\setup.py
py36 inst-nodeps: C:\python_test\sampleproject\.tox\.tmp\package\1\sampleproject-2.0.0.zip
py36 installed: atomicwrites==1.4.0,attrs==20.3.0,build==0.3.1.post1,check-manifest==0.46,colorama==0.4.4,flake8==3.9.0,importlib-metadata==3.7.3,iniconfig==1.1.1,mccabe==0.6.1,packaging==20.9,pep517==0.10.0,peppercorn==0.6,pluggy==0.13.1,py==1.10.0,pycodestyle==2.7.0,pyflakes==2.3.0,pyparsing==2.4.7,pytest==6.2.2,sampleproject @ file:///C:/python_test/sampleproject/.tox/.tmp/package/1/sampleproject-2.0.0.zip,toml==0.10.2,typing-extensions==3.7.4.3,zipp==3.4.1
py36 run-test-pre: PYTHONHASHSEED='560'
py36 run-test: commands[0] | check-manifest --ignore 'tox.ini,tests/**'
lists of files in version control and sdist do not match!
missing from sdist:
  .idea/inspectionProfiles/Project_Default.xml
  .idea/inspectionProfiles/profiles_settings.xml
  .idea/misc.xml
  .idea/modules.xml
  .idea/sampleproject.iml
  .idea/vcs.xml
ERROR: InvocationError for command 'C:\python_test\sampleproject\.tox\py36\Scripts\check-manifest.EXE' --ignore 'tox.ini,tests/**' (exited with code 1)
py37 create: C:\python_test\sampleproject\.tox\py37
ERROR: InterpreterNotFound: python3.7
py38 inst-nodeps: C:\python_test\sampleproject\.tox\.tmp\package\1\sampleproject-2.0.0.zip
py38 installed: atomicwrites==1.4.0,attrs==20.3.0,build==0.3.1.post1,check-manifest==0.46,colorama==0.4.4,flake8==3.9.0,iniconfig==1.1.1,mccabe==0.6.1,packaging==20.9,pep517==0.10.0,peppercorn==0.6,pluggy==0.13.1,py==1.10.0,pycodestyle==2.7.0,pyflakes==2.3.0,pyparsing==2.4.7,pytest==6.2.2,sampleproject @ file:///C:/python_test/sampleproject/.tox/.tmp/package/1/sampleproject-2.0.0.zip,toml==0.10.2
py38 run-test-pre: PYTHONHASHSEED='560'
py38 run-test: commands[0] | check-manifest --ignore 'tox.ini,tests/**'
lists of files in version control and sdist do not match!
missing from sdist:
  .idea/inspectionProfiles/Project_Default.xml
  .idea/inspectionProfiles/profiles_settings.xml
  .idea/misc.xml
  .idea/modules.xml
  .idea/sampleproject.iml
  .idea/vcs.xml
ERROR: InvocationError for command 'C:\python_test\sampleproject\.tox\py38\Scripts\check-manifest.EXE' --ignore 'tox.ini,tests/**' (exited with code 1)
py39 create: C:\python_test\sampleproject\.tox\py39
py39 installdeps: check-manifest >= 0.42, flake8, pytest
py39 inst: C:\python_test\sampleproject\.tox\.tmp\package\1\sampleproject-2.0.0.zip
_____________________________________________________________________________________________________________ summary ______________________________________________________________________________________________________________
ERROR:   py36: commands failed
ERROR:  py37: InterpreterNotFound: python3.7
ERROR:   py38: commands failed
  py39: commands succeeded
Traceback (most recent call last):
  File "C:\Users\111-g\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\py\_error.py", line 66, in checked_call
    return func(*args, **kwargs)
OSError: [Errno 22] Invalid argument: 'C:\\Users\\111-g\\AppData\\Local\\Microsoft\\WindowsApps\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\python.exe'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.752.0_x64__qbz5n2kfra8p0\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.752.0_x64__qbz5n2kfra8p0\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\111-g\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\tox\__main__.py", line 4, in <module>
    tox.cmdline()
  File "C:\Users\111-g\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\tox\session\__init__.py", line 44, in cmdline
    main(args)
  File "C:\Users\111-g\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\tox\session\__init__.py", line 69, in main
    exit_code = session.runcommand()
  File "C:\Users\111-g\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\tox\session\__init__.py", line 197, in runcommand
    return self.subcommand_test()
  File "C:\Users\111-g\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\tox\session\__init__.py", line 225, in subcommand_test
    run_sequential(self.config, self.venv_dict)
  File "C:\Users\111-g\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\tox\session\commands\run\sequential.py", line 18, in run_sequential
    installpkg(venv, venv.package)
  File "C:\Users\111-g\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\tox\session\commands\run\sequential.py", line 46, in installpkg
    venv.installpkg(path, action)
  File "C:\Users\111-g\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\tox\venv.py", line 396, in installpkg
    self.install_pkg(sdistpath, action, "inst")
  File "C:\Users\111-g\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\tox\venv.py", line 376, in install_pkg
    self.finish()
  File "C:\Users\111-g\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\tox\venv.py", line 318, in finish
    live_config = self._getliveconfig()
  File "C:\Users\111-g\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\tox\venv.py", line 289, in _getliveconfig
    base_resolved_python_sha256 = getdigest(base_resolved_python_path)
  File "C:\Users\111-g\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\tox\venv.py", line 669, in getdigest
    return path.computehash("sha256")
  File "C:\Users\111-g\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\py\_path\local.py", line 240, in computehash
    f = self.open('rb')
  File "C:\Users\111-g\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\py\_path\local.py", line 369, in open
    return py.error.checked_call(open, self.strpath, mode)
  File "C:\Users\111-g\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\py\_error.py", line 86, in checked_call
    raise cls("%s%r" % (func.__name__, args))
py.error.ENOTDIR: [Not a directory]: open('C:\\Users\\111-g\\AppData\\Local\\Microsoft\\WindowsApps\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\python.exe', 'rb')

C:\python_test\sampleproject>

**Comment by ross-spencer** [comment]
@ws2003gq did you find a solution?

**Comment by ross-spencer** [comment]
@ws2003gq for what it's worth - Windows used the same suffix on my machine "qbz5n2kfra8p0" this led me here. I took another's advice and decided against the Windows APP store version of Python. For whatever reason they had to customize parts of Python it was too much hassle. I was able to install Python locally via: https://www.python.org/downloads/windows/ - if that still doesn't work check `where python` and ensure that your own install comes first in the list (i.e. not the Windows one). Update the path with `setx path "...;C:\my\path\to\python"` if necessary - again, making sure the entry appears before `..\AppData\Local\Microsoft\WindowsApps` or deleting Python components including PIP from there. 

All that being said - I suspect this isn't really a question for `pypa` so perhaps this ticket can be closed?
